### DISCUSSION TITLE
Does next-intl support a flat JSON structure?

### DISCUSSION THREAD

**Discussion by catthiba** [amannn/next-intl#570] [Q&A] [answered]
Hi!

I am using a translation management tool that requires a flat JSON structure. Does next-intl support that?

I want this structure:
```
{
  "common.about-us": "About us",
  "common.contact": "Contact",
  "common.members": "Members"
}
```

And not his: 
```
{
"common": {
   "about-us": "About us",
   "contact": "Contact",
   "members": "Members"
   }
}
```

**Comment by amannn** [comment]
Sure, you can map the flat structure to the nested one before passing the messages to `next-intl`: https://github.com/amannn/next-intl/discussions/148

**Reply by catthiba** [reply]
Thanks for the quick response! 

Is there no other way? If not, when is the best time to map it?

**Reply by amannn** [reply, marked as answer]
You can do that before passing the messages to `next-intl`: [Providing messages](https://next-intl-docs.vercel.app/docs/configuration#messages).

**Reply by catthiba** [reply]
Thank you!

Since I am using Next.js 13 I had to map the messages both for Client components with the `NextIntlClientProvider`, and for Server components with the `getRequestConfig` function written in the i18n.ts configuration file. 

Seems to work fine! 😄
